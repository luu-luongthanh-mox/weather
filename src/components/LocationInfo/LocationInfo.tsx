import React from "react";
import { useSelector } from "react-redux";

import { StateStatus } from "@models/State";
import { getForecastWeather } from "@stores/forecastWeather/selector";

import styles from "./locationInfo.module.css";

export const LocationInfo = () => {
  const { loading, data } = useSelector(getForecastWeather);
  if (loading === StateStatus.loading) {
    return <p>Loading data...</p>;
  }
  if (data?.title) {
    return (
      <div className={styles.container}>
        <p className={styles.cityName}>{data.title}</p>
        <p className={styles.type}>{data.location_type}</p>
        <p>timezone: {data.timezone}</p>
      </div>
    );
  }

  return null;
};
