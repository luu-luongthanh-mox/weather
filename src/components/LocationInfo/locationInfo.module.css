.cityName {
  font-size: 1.5rem;
  font-weight: bold;
  margin-bottom: 0px;
  color: #0062cc;
}

.container {
  margin-top: 20px;
}

.type {
  font-weight: bold;
}
