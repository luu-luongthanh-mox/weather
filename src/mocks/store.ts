import { RootState } from "@stores";
import { StateStatus } from "@models/State";

import { forecastWeatherItemMock } from "./forecastWeather";

export const storeMock: RootState = {
  forecastWeather: {
    data: {
      consolidated_weather: [forecastWeatherItemMock],
      time: "2020-12-06T10:54:45.650562+08:00",
      sun_rise: "2020-12-06T06:50:25.886047+08:00",
      sun_set: "2020-12-06T17:39:03.353222+08:00",
      timezone_name: "LMT",
      title: "Hong Kong",
      location_type: "City",
      woeid: 2165352,
      latt_long: "22.411200,114.153999",
      timezone: "Asia/Hong_Kong",
    },
    loading: StateStatus.idle,
    currentRequestId: undefined,
  },
  location: {
    loading: StateStatus.idle,
    selected: null,
    searchResult: [],
    currentRequestId: undefined,
  },
};
