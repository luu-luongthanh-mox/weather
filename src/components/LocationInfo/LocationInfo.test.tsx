import React from "react";
import { Provider } from "react-redux";
import createMockStore from "redux-mock-store";
import { render, screen } from "@testing-library/react";
import { getDefaultMiddleware } from "@reduxjs/toolkit";

import { storeMock } from "@mocks/store";
import { StateStatus } from "@models/State";

import { LocationInfo } from "./LocationInfo";

const mockStore = createMockStore(getDefaultMiddleware());
const store = mockStore(storeMock);

describe("Component: LocationInfo", () => {
  it("renders data on screen", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <LocationInfo />
      </Provider>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders city name", () => {
    render(
      <Provider store={store}>
        <LocationInfo />
      </Provider>
    );
    const linkElement = screen.getByText(/Hong Kong/i);
    expect(linkElement).toBeInTheDocument();
  });

  it("renders city type", () => {
    render(
      <Provider store={store}>
        <LocationInfo />
      </Provider>
    );
    const linkElement = screen.getByText(/City/i);
    expect(linkElement).toBeInTheDocument();
  });

  it("renders city timezone", () => {
    render(
      <Provider store={store}>
        <LocationInfo />
      </Provider>
    );
    const linkElement = screen.getByText(/timezone: Asia\/Hong_Kong/i);
    expect(linkElement).toBeInTheDocument();
  });

  it("renders loading", () => {
    const newMock = { ...storeMock };
    newMock.forecastWeather.loading = StateStatus.loading;
    const localStore = mockStore(newMock);
    render(
      <Provider store={localStore}>
        <LocationInfo />
      </Provider>
    );
    const linkElement = screen.getByText(/Loading data/i);
    expect(linkElement).toBeInTheDocument();
  });
});
