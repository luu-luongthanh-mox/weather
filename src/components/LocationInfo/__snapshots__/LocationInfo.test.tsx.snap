// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: LocationInfo renders data on screen 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <p
      class="cityName"
    >
      Hong Kong
    </p>
    <p
      class="type"
    >
      City
    </p>
    <p>
      timezone: Asia/Hong_Kong
    </p>
  </div>
</DocumentFragment>
`;
